### PR TITLE
PHOENIX-6758 During HBase 2 upgrade Phoenix Self healing task fails to create server side connection before reading SYSTEM.TASK

### PR DESCRIPTION
**Pull request by virajjasani**

### CODE REVIEW

**Comment by virajjasani** [comment]
5.1 backport of PR #1477

**Comment by virajjasani** [comment]
All tests passed on the build https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1478/1/testReport/

File: phoenix-core/src/main/java/org/apache/phoenix/query/AdminUtilWithFallback.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
+package org.apache.phoenix.query;
```
Comment (gjacoby126) [inline phoenix-core/src/main/java/org/apache/phoenix/query/AdminUtilWithFallback.java:1]: Please add an Apache license

**Review by gjacoby126 — CHANGES_REQUESTED** [review]

**Review by gjacoby126 — APPROVED** [review]
+1

**Pull request merged**